New proposition for the tutorials section layout (#1779)
ansys/pydpf-core: doc/source/user_guide/tutorials/index.rst

Tutorials
---------

What you need to know
+++++++++++++++++++++

These tutorials go through the steps required to access, analyze,
and transform simulation data using  PyDPF-Core.

Tutorials are more substantive and complex than examples. They are designed to teach how to perform a task and understand
the underlying concepts, providing detailed explanations at each stage, whereas examples showcase end-to-end specific processes.

DPF interacts with data stored in DPF data structures.
These DPF data structures are generated automatically when reading data from results file (`ref_to_section`) but they can also be generated from scratch (`ref_to_section`) when no supported result file is available.

Overview
++++++++

> There are four main steps to transform simulation data into output data that can
> be used to visualize and analyze simulation results:

Author: ```suggestion
 There are different kind of actions you can perform with PyDPF-Core: from extracting result from result files, to visualization of extracted data, to filtering, transformation of the data, and much more. 

Nevertheless,  we can summarize in four main steps the process of transforming simulation data into post-processed output data :
```

Tutorials
---------

> What do you need to know
> ++++++++++++++++++++++++

Reviewer: ```suggestion
What you need to know
+++++++++++++++++++++
```

> This tutorials will guide you through the steps required to access, analyze,
> and transform simulation data using  PyDPF-Core.

Reviewer: ```suggestion
These tutorials go through the steps required to access, analyze,
and transform simulation data using  PyDPF-Core.
```

> The tutorials are more substantive and complex than examples found in the standard
> documentation set. They are designed to teach how to perform a task and understand
> some concepts, providing detailed explanations at each stage.

Reviewer: ```suggestion
Tutorials are more substantive and complex than examples. They are designed to teach how to perform a task and understand
the underlying concepts, providing detailed explanations at each stage, whereas examples showcase end-to-end specific processes.
```

Reviewer: Not sure if tutorials can be said to be more `complex` than examples.

Author: more elaborated ?

> DPF interacts with results data that have to be stored in some kind  of structure.
> They can be automatically generated from the results file (detailed in the access
> and extract results part ) or generated from scripting.

Reviewer: ```suggestion
DPF interacts with data stored in DPF data structures.
These are generated automatically when reading data from results file (`ref_to_section`) but can also be generated from scratch (`ref_to_section`).
```

Overview
++++++++

There are four main steps to transform simulation data into output data that can
be used to visualize and analyze simulation results:

    :bdg-dark-line:`Define simulation data` :octicon:`arrow-right;1em` :bdg-dark-line:`Store input data in DPF` :octicon:`arrow-right;1em` :bdg-dark-line:`Transform available data` :octicon:`arrow-right;1em` :bdg-dark-line:`Extract data`

> The following tutorials will guide through those steps. The tutorials explains separately
> the used methods for files of each acceptable solver types.

Reviewer: ```suggestion
The following tutorials go through each of these steps. Each tutorial presents code snippets specific to each supported file format.
```

> .. topic:: Importing and opening results files

Reviewer: ```suggestion
.. topic:: Extract data from result files
```

Author: I don't know if using "extract " here is a great idea. Because in the other section we talk about extracting (like if I want to have a displacement result). So here is it actually extract a result?

    .. grid:: 3

>         .. grid-item-card:: I have one file

Reviewer: ```suggestion
        .. grid-item-card:: Extract data from a single file
```

           :link: ref_tutorials_model
           :link-type: ref
           :text-align: center

>            Import and open a single result file with PyDPF-Core.

Reviewer: ```suggestion
           Extract data from a single result file with PyDPF-Core.
```

>         .. grid-item-card:: I have more than one file

Reviewer: ```suggestion
        .. grid-item-card:: Extract data from a combination of files
```

Reviewer: And the idea would be to also add a tutorial for:
- when you have accessory files
- when you have upstream result files
- and so on

Author: Yes. If you have more than one file the tutorial will talk about these cases
